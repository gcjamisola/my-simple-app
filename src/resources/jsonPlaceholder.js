require('es6-promise').polyfill();
require('isomorphic-fetch');

const rGetPhotos = () => {
	return fetch('https://jsonplaceholder.typicode.com/photos')
		.then((response) => response.json());
};

const rGetUsers = (id = '') => {
	return fetch(`https://jsonplaceholder.typicode.com/users/${id}`)
		.then((response) => response.json());
};

const rAddUser = (data) => {
	return fetch('https://jsonplaceholder.typicode.com/users', {
		method: 'POST',
		body: JSON.stringify(data),
		headers: {
			'Content-type': 'application/json; charset=UTF-8'
		}
	})
		.then((response) => response.json());
};

const rSendEmail = (data) => {
	return fetch('https://mandrillapp.com/api/1.0/messages/send.json', {
		method: 'POST',
		body: JSON.stringify(data),
		headers: {
			'Content-type': 'application/json; charset=UTF-8'
		}
	})
		.then((response) => response.json());
};

export {
	rGetPhotos,
	rGetUsers,
	rAddUser,
	rSendEmail
};