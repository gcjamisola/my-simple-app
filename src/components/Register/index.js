import React, { Component } from 'react';
import styled from 'styled-components';
import _ from 'lodash';

import Popup from '../Modal';

import { rAddUser } from '../../resources/jsonPlaceholder';
import {Form, Button} from '../../styles';

class Register extends Component {
	constructor(props) {
		super(props);

		this.state = {
			trigger: false
		};

		this._handleSubmit = this._handleSubmit.bind(this);
	}

	_handleSubmit(e) {
		e.preventDefault();

		const dataObj = _.mapValues(_.keyBy($(e.target).serializeArray(), 'name'), 'value');

		const userObj = {
			'name': dataObj.name,
			'username': dataObj.username,
			'email': dataObj.email,
			'address': {
				'street': dataObj.street,
				'suite': 'Ipsum',
				'city': dataObj.city,
				'zipcode': dataObj.zipcode,
				'geo': {
					'lat': '-37.3159',
					'lng': '81.1496'
				}
			},
			'phone': dataObj.phone,
			'website': dataObj.url,
			'company': {
				'name': 'Lorem-Ipsum-company',
				'catchPhrase': 'Multi-layered client-server neural-net',
				'bs': 'harness real-time e-markets'
			}
		};

		rAddUser(userObj)
			.then((response) => {
				// const {trigger} = this.state;
				// response ? this.setState({trigger: !trigger}) : '';
				response ? alert('User has already been added !') : '';
			});
	}

	render() {
		const RegisterContainer = styled.div`
			display: flex;
			justify-content: center;
		`;

		const {trigger} = this.state;

		return (
			<RegisterContainer>
				<Form onSubmit={this._handleSubmit}>
					<ul>
						<li>
							<label htmlFor='name'>Name</label>
							<input type='text' name='name' maxLength='100' />
							<span>Enter your full name here</span>
						</li>
						<li>
							<label htmlFor='username'>Username</label>
							<input type='text' name='username' maxLength='100' />
							<span>Enter a username</span>
						</li>
						<li>
							<label htmlFor='email'>Email</label>
							<input type='email' name='email' maxLength='100' />
							<span>Enter a valid email address</span>
						</li>
						<li>
							<label htmlFor='address'>Address</label>
							<input type='text' name='street' maxLength='100' placeholder='street'/>
							<input type='text' name='city' maxLength='100' placeholder='city'/>
							<input type='text' name='zipcode' maxLength='100' placeholder='zipcode'/>
							<span>Enter a address</span>
						</li>
						<li>
							<label htmlFor='phone'>Phone</label>
							<input type='text' name='phone' maxLength='100' />
							<span>Enter phone number</span>
						</li>
						<li>
							<label htmlFor='url'>Website</label>
							<input type='url' name='url' maxLength='100' />
							<span>Your website address (eg: http://www.google.com)</span>
						</li>
						<li>
							<Button>Submit</Button>
						</li>
					</ul>
				</Form>
				<Popup open={trigger}/>
			</RegisterContainer>
		);
	}
}

export default Register;