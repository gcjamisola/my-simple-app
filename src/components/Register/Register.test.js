import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Register from './index';
import { rAddUser } from '../../resources/jsonPlaceholder';

describe('Register component', () => {
	it('should render the component', () => {
		const component = shallow(<Register />);
		const tree = toJson(component);
		expect(tree).toMatchSnapshot();
	});

	it('should return response upon submission', () => {
		const userData = {
			'name': 'Lorem Ipsum',
			'username': 'lorem.ipsum',
			'email': 'lorem.ipsum@li.biz',
			'address': {
				'street': 'Lorem',
				'suite': 'Ipsum',
				'city': 'Dolor',
				'zipcode': '92998-3874',
				'geo': {
					'lat': '-37.3159',
					'lng': '81.1496'
				}
			},
			'phone': '1-770-736-8031 x56442',
			'website': 'loremipsum.org',
			'company': {
				'name': 'Lorem-Ipsum-company',
				'catchPhrase': 'Multi-layered client-server neural-net',
				'bs': 'harness real-time e-markets'
			}
		};

		return rAddUser(userData)
			.then((response) => {
				expect(response).toBeDefined();
			});
	});
});