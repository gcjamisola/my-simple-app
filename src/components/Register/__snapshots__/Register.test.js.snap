// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component should render the component 1`] = `
<styled.div>
  <styled.form
    onSubmit={[Function]}
  >
    <ul>
      <li>
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          maxLength="100"
          name="name"
          type="text"
        />
        <span>
          Enter your full name here
        </span>
      </li>
      <li>
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          maxLength="100"
          name="username"
          type="text"
        />
        <span>
          Enter a username
        </span>
      </li>
      <li>
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          maxLength="100"
          name="email"
          type="email"
        />
        <span>
          Enter a valid email address
        </span>
      </li>
      <li>
        <label
          htmlFor="address"
        >
          Address
        </label>
        <input
          maxLength="100"
          name="street"
          placeholder="street"
          type="text"
        />
        <input
          maxLength="100"
          name="city"
          placeholder="city"
          type="text"
        />
        <input
          maxLength="100"
          name="zipcode"
          placeholder="zipcode"
          type="text"
        />
        <span>
          Enter a address
        </span>
      </li>
      <li>
        <label
          htmlFor="phone"
        >
          Phone
        </label>
        <input
          maxLength="100"
          name="phone"
          type="text"
        />
        <span>
          Enter phone number
        </span>
      </li>
      <li>
        <label
          htmlFor="url"
        >
          Website
        </label>
        <input
          maxLength="100"
          name="url"
          type="url"
        />
        <span>
          Your website address (eg: http://www.google.com)
        </span>
      </li>
      <li>
        <styled.button>
          Submit
        </styled.button>
      </li>
    </ul>
  </styled.form>
  <Popup
    open={false}
  />
</styled.div>
`;
